### PR TITLE
Added actions, reducers & selectors to fetch the locations from the WooCommerce host.

### LINKED ISSUE #14266: Infrastructure: Shipping Zones: Add action that gets the available shipping locations (continents, countries & states) from merchant host endpoint

### PR DESCRIPTION
**Pull request by DanReyLop**
Fixes #14266

This PR adds logic to fetch locations data from the WooCommerce server. It uses the new `v3/data/continents` endpoint ([from here](https://github.com/woocommerce/wc-api-dev/commit/c939719f286b71708a7637f3f0e65faac27acb79#diff-5ddc439b854b48ebdb5d004187141e5f)). Since each continent already is returned with all the info for all its countries, and every country is returned with all the info for all its states, there's no need to hit any other endpoint.

The data is stored as it comes from the server, but I've added a few selectors to "query" it. Since there are quite a few countries in the world, maybe I've been too negligent with performance. When this is hooked to an actual UI, I'll measure and if needed optimize the data structures or add memoizing to the selectors.

There is no way to test it in the UI, so just run the tests.

### CODE REVIEW

File: client/extensions/woocommerce/state/sites/locations/selectors.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { LOADING } from 'woocommerce/state/constants';
+
+/**
+ * @param {Object} state Whole Redux state tree
+ * @param {Number} [siteId] Site ID to check. If not provided, the Site ID selected in the UI will be used
+ * @return {Array} The locations tree, as retrieved from the server. It can also be the string "LOADING"
+ * if the locations are currently being fetched, or a "falsy" value if that haven't been fetched at all.
+ */
+export const getRawLocations = ( state, siteId = getSelectedSiteId( state ) ) => {
```
Comment (belcherj) [inline client/extensions/woocommerce/state/sites/locations/selectors.js]: do you need the Raw here?

Comment (belcherj) [inline client/extensions/woocommerce/state/sites/locations/selectors.js]: Is this being used somewhere else?  From this PR it is not evident why it is being exported

Comment (DanReyLop) [inline client/extensions/woocommerce/state/sites/locations/selectors.js]: You're right, it doesn't need to be exported. Fixed.

File: client/extensions/woocommerce/state/sites/locations/README.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+				{
+					code: "EG",
+					name: "Egypt",
+					states: [],
+				},
+				{ ... }
+			],
+		},
+		{
+			code: "AM",
```
Comment (justinshreve) [inline client/extensions/woocommerce/state/sites/locations/README.md]: Minor but this should be `NA` and `North America`.

File: client/extensions/woocommerce/state/sites/locations/test/reducer.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+				name: 'Africa',
+				countries: [
+					{
+						code: 'SA',
+						name: 'South Africa',
+						states: [],
+					},
+				],
+			},
+			{
```
Comment (justinshreve) [inline client/extensions/woocommerce/state/sites/locations/test/reducer.js]: `NA` and `North America`

File: client/extensions/woocommerce/state/sites/locations/test/selectors.js [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+	getContinents,
+	getCountries,
+	getStates,
+	hasStates,
+} from '../selectors';
+import { LOADING } from 'woocommerce/state/constants';
+
+const locations = [
+	{
+		code: 'AM',
```
Comment (justinshreve) [inline client/extensions/woocommerce/state/sites/locations/test/selectors.js]: `NA` and `North America`

File: client/extensions/woocommerce/state/sites/locations/test/selectors.js [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+
+		it( 'should return false if the country does not exist', () => {
+			expect( hasStates( loadedState, 'XX' ) ).to.be.false;
+		} );
+
+		it( 'should return false if the country does not have states', () => {
+			expect( hasStates( loadedState, 'SA' ) ).to.be.false;
+		} );
+
+		it( 'should return true if the countri has states', () => {
```
Comment (justinshreve) [inline client/extensions/woocommerce/state/sites/locations/test/selectors.js]: Typo: `country`

**Comment by DanReyLop** [comment]
> Overall looks good to me. The main thing is replacing AM/America in the docs and tests with NA and North America which match what would be returned.

🤦‍♂️ fixed. It was just for an example, but I should have looked at the real values anyway :)

**Comment by DanReyLop** [comment]
@belcherj @justinshreve Can I please get another quick 👀? I've fixed the comments you had.

**Pull request merged**